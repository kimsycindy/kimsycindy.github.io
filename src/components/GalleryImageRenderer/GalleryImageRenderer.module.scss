.GalleryImageRenderer {
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  cursor: pointer;
  outline: none;
  box-shadow: none;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.GalleryImageRenderer:hover {
  filter: brightness(150%);
}
