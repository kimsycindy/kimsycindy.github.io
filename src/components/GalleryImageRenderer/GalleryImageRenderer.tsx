import React from 'react'
import Img, { FluidObject } from 'gatsby-image'
import { RenderImageProps } from 'react-photo-gallery'

import styles from './GalleryImageRenderer.module.scss'

const GalleryImageRenderer = ({
  index,
  photo,
  margin,
  onClick,
}: RenderImageProps) => {
  const fluidObject: FluidObject = {
    aspectRatio: photo.width / photo.height,
    src: photo.src,
    srcSet: photo.srcSet?.toString() ?? '',
    sizes: photo.sizes?.toString() ?? '',
  }

  return (
    <button
      type="button"
      key={photo.key}
      onClick={event => {
        if (onClick) {
          onClick(event, { ...photo, index })
        }
      }}
      style={{ margin, width: photo.width, height: photo.height }}
      className={styles.GalleryImageRenderer}
    >
      <Img fluid={fluidObject} />
    </button>
  )
}

export default GalleryImageRenderer
