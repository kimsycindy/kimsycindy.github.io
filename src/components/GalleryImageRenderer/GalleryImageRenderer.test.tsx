import React from 'react'
import renderer from 'react-test-renderer'

import GalleryImageRenderer from './GalleryImageRenderer'

describe('GalleryImageRenderer', () => {
  it('renders correctly', () => {
    const fluid = {
      src: 'https://cindy.kim',
      srcSet: 'elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w',
      sizes: '(max-width: 600px) 480px, 800px',
      width: 500,
      height: 500,
    }

    const tree = renderer
      .create(
        <GalleryImageRenderer
          index={0}
          photo={fluid}
          onClick={jest.fn()}
          direction="row"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
