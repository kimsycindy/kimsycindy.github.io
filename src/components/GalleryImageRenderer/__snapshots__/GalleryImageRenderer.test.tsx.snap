// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryImageRenderer renders correctly 1`] = `
<button
  className="GalleryImageRenderer"
  onClick={[Function]}
  style={
    Object {
      "height": 500,
      "margin": undefined,
      "width": 500,
    }
  }
  type="button"
>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "maxHeight": null,
        "maxWidth": null,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      aria-hidden={true}
      style={
        Object {
          "paddingBottom": "100%",
          "width": "100%",
        }
      }
    />
    <picture>
      <source
        sizes="(max-width: 600px) 480px, 800px"
        srcSet="elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w"
      />
      <img
        alt=""
        loading="lazy"
        onLoad={[Function]}
        sizes="(max-width: 600px) 480px, 800px"
        src="https://cindy.kim"
        srcSet="elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source srcset=\\"elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w\\" sizes=\\"(max-width: 600px) 480px, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 600px) 480px, 800px\\" srcset=\\"elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w\\" src=\\"https://cindy.kim\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
</button>
`;
